### PR TITLE
cleanup example/json

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
I did *NOT* do a good job reviewing your PR, @subfuzion .

I also forgot to add the workflow change to catch issues.

My bad...

File: .github/workflows/verify_examples.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        example_directory: ["examples/hello"]
+        example_directory: ["examples/hello", "examples/json"]
```
Comment (subfuzion) [inline .github/workflows/verify_examples.yml:21]: good catch!

**Review by subfuzion — APPROVED** [review]

**Comment by kevmoo** [comment]
PTAL @subfuzion

**Comment by subfuzion** [comment]
> I guess you wanted to keep this crazy simple?

It's fine, but now the example is sooooo much about JSON ceremony ... a bit outsized relative to the function, which is what we're trying to highlight. I suppose it's more realistic though. I had though about subclassing in the test and adding the equals and hashcode overrides there, but you kept pinging if I was ready lol...

**Comment by subfuzion** [comment]
> Again. This is a PR. Nothing is committed. Having a discussion here...

Not sure what you mean by "Again." I'm discussing with you why I'm not so enthusiastic about the change, but if you feel strongly about it, that's fine.

**Review by subfuzion — APPROVED** [review]
Thanks! Next time, let me clean my own code!

File: examples/json/lib/functions.dart [hunk 2]
```diff
@@ -26,7 +24,19 @@ class GreetingResponse {
 
   GreetingResponse({this.salutation, this.name});
 
+  factory GreetingResponse.fromJson(Map<String, dynamic> json) =>
```
Comment (subfuzion) [inline examples/json/lib/functions.dart:27]: README should get updated or else we should remove the "The full code is shown below" part

**Pull request merged**